core/math/quaternion.h

```diff
@@ -139,7 +139,7 @@ struct [[nodiscard]] Quaternion {
 #ifdef MATH_CHECKS
 		ERR_FAIL_COND_MSG(p_v0.is_zero_approx() || p_v1.is_zero_approx(), "The vectors must not be zero.");
 #endif
-		constexpr real_t ALMOST_ONE = 1.0f - (real_t)CMP_EPSILON;
+		constexpr real_t ALMOST_ONE = 1.0f - (real_t)0.00000025;
```

Comment:
With 32-bit floats, this is 4 ULPs away from `1.0`, which seems reasonable. However, I would go even further when compiling with doubles. The value in my suggestion is 90 ULPs away from `1.0`, and while we could add one more `9` to get it 9 ULPs away, IMO it's generally safer to be a bit more tolerant with doubles since other issues may pop up so close to coordinate singularities. Btw, since we are not using `CMP_EPSILON` anymore, why not just write out the value?

```suggestion
#ifdef REAL_T_IS_DOUBLE
		constexpr real_t ALMOST_ONE = 0.99999999999999;
#else
		constexpr real_t ALMOST_ONE = 0.99999975f;
#endif
```